docs/source/x_ext.rst

```diff
@@ -570,13 +576,10 @@ then it shall issue no further memory request transactions for the same instruct
 be issued by a |coprocessor| for the same instruction (``id``). A sequence of memory request transactions therefore does not necessarily end with a transaction with ``last`` = 1.
 
 A |coprocessor| shall never initiate a memory request transaction(s) for offloaded non-accepted instructions.
-A |coprocessor| shall never initiate a memory request transaction(s) for offloaded accepted non-load/store instructions (``loadstore`` = 0).
-A |coprocessor| shall never initiate a non-speculative memory request transaction(s) for offloaded accepted load/store instructions before receiving the commit transaction.
-A |coprocessor| may initiate a speculative memory request transaction(s) for offloaded accepted load/store instructions before receiving the commit transaction.
-A |coprocessor| may initiate a (speculative or non-speculative) memory request transaction(s) for offloaded accepted load/store instructions in the same cycle as receiving ``commit_kill`` = 1.
-A |coprocessor| may initiate a (speculative or non-speculative) memory request transaction(s) for offloaded accepted load/store instructions in the same cycle as receiving ``commit_kill`` = 0.
-A |coprocessor| shall never initiate a (speculative or non-speculative) memory request transaction(s) for offloaded accepted load/store instructions after receiving ``commit_kill`` = 1 via the commit transaction.
-A |coprocessor| shall initiate a (speculative or non-speculative) memory request transaction(s) for offloaded accepted load/store instructions that receive ``commit_kill`` = 0 via the commit transaction.
+A |coprocessor| shall never initiate a memory request transaction(s) for offloaded non-load/store instructions (``loadstore`` = 0).
+A |coprocessor| shall never initiate a non-speculative memory request transaction(s) unless on or after a cycle of receiving a commit transaction with ``commit_kill`` = 0.
+A |coprocessor| shall never initiate a speculative memory request transaction(s) after receiving ``commit_kill`` = 1 via the commit transaction.
+A |coprocessor| shall initiate memory request transaction(s) for offloaded accepted load/store instructions that receive ``commit_kill`` = 0 via the commit transaction.
```

Comment:
```suggestion
A |coprocessor| shall initiate a non-speculative memory request transaction(s) for offloaded accepted load/store instructions that receive ``commit_kill`` = 0 via the commit transaction.
```
Does this suggestion make sense?

I was thinking about the following case:
A coproc does a speculative mem req while waiting to be committed.
It gets informed about an exception (e.g. from PMA).
It realizes there is no more work it can do, since there is an exception.
But after it is committed, then it should be obliged to retry the memory request, but as non-speculative, to cause the actual exception in the cpu?

It is not sufficient to allow for a coproc to respond to being commited simply with a non-speculative.

(It is permitted to do speculative accesses even on cycles after being committed, right?)

--------

Also, I can't see that "speculative memory request" is defined?
I assumed it means that the cpu will give the transaction a try, but there will be no side-effects?
Will the coproc still be informed if the transaction _would_ have caused an exception?
Will the coproc still get the read data if the transaction goes through?
Will the transaction not reach the bus, but only be tested for PMP etc?

Reply:
> If the coprocessor causes an exception, then that is it, i.e. no need to replay anything.

Alright, I almost see it now.
But the whole process is still inconsistent in my mind.

Does this mean that the core will not do any of its usual exception-related side-effects when a memory request causes an exception, but it will wait until the coproc signals exception on the result interface?

What about debug?
That cannot be signaled in the result interface.
So in that case, the mem resp shall be acted upon immediately instead of waiting for result interface?
Shall there still be a result interface transaction?
Is it symmetrical to the exc signaling or not.

What is semantically the point of speculative memory requests?
It seems like everything behaves exactly the same, but with one signal high and with less strict behavioral requirements.
I originally expected it to mean something like 1) no side-effects taken by core, or 2) speculative can only be reads.
What am I missing?

_"A synchronous exception will lead to a trap in CPU unless the corresponding instruction is killed."_ should this sentence really be present in both the memory interface description and in the result interface description, or only the latter?